app/lib/frontend/handlers.dart

```diff
@@ -435,12 +435,24 @@ Future<shelf.Response> _packagesHandlerHtmlV2(
     shelf.Request request, String platform) async {
   final int page = _pageFromUrl(request.url);
   final int offset = PackageLinks.resultsPerPage * (page - 1);
-  final queryText = request.requestedUri.queryParameters['q'] ?? '';
-  // TODO: handle package-prefix
+  String queryText = request.requestedUri.queryParameters['q'] ?? '';
+  String packagePrefix = request.url.queryParameters['pkg-prefix'];
+
+  if (queryText != null) {
+    final Match pkgMatch = _packageRegexp.firstMatch(queryText);
+    if (pkgMatch != null) {
+      // only if query attribute was not specified
```

Comment:
Maybe remove this comment and say rather the following (at line 441):
```
// If queryText is of the form "a package:foo/a/b/c.dart z" we try to set
//     pkg-prefix=foo
//     queryText= a /a/b/c.dart z
```
Even though I'm not quite sure if that's what's happening further down with `queryText.replaceFirst`, since the regexp is defined in a weired way:

```dart
final _packageRegexp = new RegExp('package:([_a-z0-9]+)\\*?', caseSensitive: false);
```

Does this mean that `package:foo` must be followed by 0 or more backslashes, which are then marked as optional with the question mark? Somehow I think this should be rather just `package:([_a-z0-9]+)`.